hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/common/detailslayout/AbstractDistributionSetDetails.java

```diff
@@ -0,0 +1,214 @@
+import org.eclipse.hawkbit.ui.management.event.DistributionTableEvent;
+import org.eclipse.hawkbit.ui.management.state.ManagementUIState;
+import org.eclipse.hawkbit.ui.utils.I18N;
+import org.eclipse.hawkbit.ui.utils.UIComponentIdProvider;
+import org.eclipse.hawkbit.ui.utils.UINotification;
+import org.vaadin.spring.events.EventBus.UIEventBus;
+import org.vaadin.spring.events.EventScope;
+import org.vaadin.spring.events.annotation.EventBusListenerMethod;
+
+import com.vaadin.ui.Button.ClickEvent;
+import com.vaadin.ui.Label;
+import com.vaadin.ui.TabSheet;
+import com.vaadin.ui.UI;
+import com.vaadin.ui.VerticalLayout;
+import com.vaadin.ui.Window;
+
+/**
+ * Abstract class which contains common code for Distribution Set Details
+ *
+ */
+public abstract class AbstractDistributionSetDetails
+        extends AbstractNamedVersionedEntityTableDetailsLayout<DistributionSet> {
+
+    private static final long serialVersionUID = 1L;
+
+    private final DistributionAddUpdateWindowLayout distributionAddUpdateWindowLayout;
+
+    private final DistributionSetMetadatadetailsLayout dsMetadataTable;
+
+    private final UINotification uiNotification;
+
+    private final transient DistributionSetManagement distributionSetManagement;
+
+    private final DsMetadataPopupLayout dsMetadataPopupLayout;
+
+    private final DistributionTagToken distributionTagToken;
+
+    private final SoftwareModuleDetailsTable softwareModuleDetailsTable;
+
+    private VerticalLayout softwareModuleTab;
+
+    protected AbstractDistributionSetDetails(final I18N i18n, final UIEventBus eventBus,
+            final SpPermissionChecker permissionChecker, final ManagementUIState managementUIState,
+            final DistributionAddUpdateWindowLayout distributionAddUpdateWindowLayout,
+            final DistributionSetManagement distributionSetManagement,
+            final DsMetadataPopupLayout dsMetadataPopupLayout, final EntityFactory entityFactory,
+            final UINotification uiNotification, final TagManagement tagManagement,
+            final SoftwareModuleDetailsTable softwareModuleDetailsTable) {
+        super(i18n, eventBus, permissionChecker, managementUIState);
+        this.distributionAddUpdateWindowLayout = distributionAddUpdateWindowLayout;
+        this.uiNotification = uiNotification;
+        this.distributionSetManagement = distributionSetManagement;
+        this.dsMetadataPopupLayout = dsMetadataPopupLayout;
+        this.distributionTagToken = new DistributionTagToken(permissionChecker, i18n, uiNotification, eventBus,
+                managementUIState, tagManagement, distributionSetManagement);
+        this.softwareModuleDetailsTable = softwareModuleDetailsTable;
+
+        dsMetadataTable = new DistributionSetMetadatadetailsLayout(i18n, permissionChecker, distributionSetManagement,
+                dsMetadataPopupLayout, entityFactory, uiNotification);
+        createSoftwareModuleTab();
```

Comment:
Is the same here. See @michahirsch  comment 
below. "Constructor should not call overridables see: http://www.javapractices.com/topic/TopicAction.do?Id=215".

The `createSoftwareModuleTab ` is final but not the `createTabLayout`